[dynamo] Provide helper functions for guard filter hook

anijain2305 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #155083

Collection of ready-made guard filters. One issue is that they are not composable - `filter1(filter2(guard))`. On the other hand, they are easy to use.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

anijain2305 committed:
[dynamo] Provide helper functions for guard filter hook

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo] Provide helper functions for guard filter hook"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo] Provide helper functions for guard filter hook"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo] Provide helper functions for guard filter hook"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo] Provide helper functions for guard filter hook"


Collection of ready-made guard filters. One issue is that they are not composable - `filter1(filter2(guard))`. On the other hand, they are easy to use.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

williamwen42 commented on test/dynamo/test_misc.py:
```diff
@@ -12286,6 +12286,111 @@ def fn(x, y):
+        x = torch.rand(4, 8)
+        opt_mod(x)
+
+        GLOBAL_INT = 1
```
change value?

anijain2305 replied:
Done

williamwen42 commented on test/dynamo/test_misc.py:
```diff
@@ -12286,6 +12286,111 @@ def fn(x, y):
+        x = torch.rand(4, 8)
+        opt_mod(x)
+
+        mod.c = 3
```
Would testing with a constant float input be more appropriate?

anijain2305 replied:
Done

williamwen42 commented on torch/compiler/__init__.py:
```diff
@@ -28,6 +28,8 @@
     "save_cache_artifacts",
     "load_cache_artifacts",
+    "skip_guard_on_inbuilt_nn_modules_unsafe",
+    "skip_guard_on_all_nn_modules_unsafe",
```
Will the other guard filters be added too?

anijain2305 replied:
Done

anijain2305 committed:
Update on "[dynamo] Provide helper functions for guard filter hook"


Collection of ready-made guard filters. One issue is that they are not composable - `filter1(filter2(guard))`. On the other hand, they are easy to use.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo] Provide helper functions for guard filter hook"


Collection of ready-made guard filters. One issue is that they are not composable - `filter1(filter2(guard))`. On the other hand, they are easy to use.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo] Provide helper functions for guard filter hook"


Collection of ready-made guard filters. One issue is that they are not composable - `filter1(filter2(guard))`. On the other hand, they are easy to use.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo] Provide helper functions for guard filter hook"


Collection of ready-made guard filters. One issue is that they are not composable - `filter1(filter2(guard))`. On the other hand, they are easy to use.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo] Provide helper functions for guard filter hook"


Collection of ready-made guard filters. One issue is that they are not composable - `filter1(filter2(guard))`. On the other hand, they are easy to use.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

zou3519 commented on docs/source/torch.compiler_api.rst:
```diff
@@ -25,3 +25,7 @@ For a quick overview of ``torch.compiler``, see :ref:`torch.compiler_overview`.
+     skip_guard_on_inbuilt_nn_modules_unsafe
+     skip_guard_on_all_nn_modules_unsafe
+     keep_tensor_guards
+     skip_guard_on_globals
```
I think the conclusions from the meeting on friday were:
1) don't expose custom guards yet,
2) Instead we are going to expose a long list of things that people want and then look at the user feedback

so I think what you have now is aligned with that discussion

anijain2305 replied:
Yes, I will add a keep_guard_on_inputs. I am not too excited about require_grad_mode and require_inference_mode. I think we should just fix the requires_inference_mode (and provide one option that is more compatible with compile). For requires_grad, we should tell the user to perhaps use `torch.no_grad`.

anijain2305 committed:
Update on "[dynamo] Provide helper functions for guard filter hook"


Collection of ready-made guard filters. One issue is that they are not composable - `filter1(filter2(guard))`. On the other hand, they are easy to use.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

anijain2305 commented:
@pytorchbot merge
